wpilibj/src/athena/cpp/lib/HALUtil.cpp

```diff
@@ -227,9 +228,10 @@ void ReportCANError(JNIEnv *env, int32_t status, int message_id) {
       break;
     }
     default: {
-      char buf[100];
-      sprintf(buf, "Fatal status code detected: %d", status);
-      env->ThrowNew(uncleanStatusExCls, buf);
+      llvm::SmallString<100> buf;
+      llvm::raw_svector_ostream oss(buf);
+      oss << "Fatal status code detected: " << status;
+      env->ThrowNew(uncleanStatusExCls, oss.str().data());
```

Comment:
I think you need to use `buf.c_str()` here and everywhere else you're doing this to ensure it's nul-terminated.